packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -0,0 +1,171 @@
+import cloudwatch = require('@aws-cdk/aws-cloudwatch');
+import events = require('@aws-cdk/aws-events');
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+import { StateGraph } from './state-graph';
+import { cloudformation } from './stepfunctions.generated';
+import { IChainable } from './types';
+
+/**
+ * Properties for defining a State Machine
+ */
+export interface StateMachineProps {
+    /**
+     * A name for the state machine
+     *
+     * @default A name is automatically generated
+     */
+    stateMachineName?: string;
+
+    /**
+     * Definition for this state machine
+     */
+    definition: IChainable;
+
+    /**
+     * The execution role for the state machine service
+     *
+     * @default A role is automatically created
+     */
+    role?: iam.Role;
+
+    /**
+     * Maximum run time for this state machine
+     *
+     * @default No timeout
+     */
+    timeoutSeconds?: number;
+}
+
+/**
+ * Define a StepFunctions State Machine
+ */
+export class StateMachine extends cdk.Construct {
```

Comment:
Is there a use base for `StateMachineRef`?

Reply:
I'm not sure there is today, but there might be once they release the feature that a State Machine can be used in a Task.